src/components/app.jsx

```diff
@@ -1,9 +1,163 @@
 import React from "react";
-import { ReactComponent as Logo } from "../images/sun.svg";
 import "../App.css";
+import { getWeather, getForecast } from '../utils.js';
+import Header from './Header';
+import WeatherCurrent from './WeatherCurrent';
+import WeatherForecast from './WeatherForecast';
 
-function App() {
-  return <h1>Weather</h1>;
+class App extends React.Component {
+  state = {
+    cityName: "",
+    currentWeather: {
+      main: "",
+      description: "",
+      temp: "",
+      windSpeed: "",
+      iconLink: ""
+    },
+    forecast: [],
+    view: "current",
+    unit: "imperial"
+  }
+
+  async componentDidMount() {
+    if (document.location.pathname !== "/") {
+      const cityName = document.location.pathname.slice(1).replace(/%20/g, " ")
+      const { unit } = this.state
+      this.setState({
+        cityName
+      })
+      this.handleWeatherState(cityName, unit)
+      this.handleForecastState(cityName, unit)
+    }
+  }
+
+  handleWeatherState = async (cityName, unit) => {
+    const response = await getWeather(cityName, unit)
```

Comment:
What would happen if there was a network error?

Reply:
I almost delayed turning this project in this morning to add error handling. Right now, if there's a network error on any of my HTTP requests, the user will have no idea that there was an error and think that the app just doesn't work. Not a great UX!

Normally, I would have wrapped my API calls in `try` / `catch`, added any error messages to state, and then rendered them in the UI. I actually began this process; however, some of the design choices I made earlier were problematic for this endeavor. For example, I stuffed both the current weather API call and the 5 day forecast API call into the same function calls. This made error handling sloppy. To fix it, I would have needed to separate them. 

If this were a real-world app, I definitely would have separated them, not only because of this messy error handling, but also because many users wouldn't need both the current weather and the 5 day forecast, so it forces the app to do more HTTP requests than necessary. I actually began this process for this project, then decided, since it was an extra feature, that maybe I should just turn it in and discuss with you. Hope that was the right choice!